lib/logger.js

```diff
@@ -112,8 +111,37 @@ class Logger {
             this._traceLogger = this._logger.child({
                 level: bunyan.TRACE
             });
-            // Set up handlers for uncaught extensions
-            this._setupRootHandlers();
+
+            // Avoid recursion if there are bugs in the logging code.
+            this.inLogger = false;
+
+            this.logUnhandledException = (err) => {
+                if (!this.inLogger) {
+                    this.inLogger = true;
+                    this.log('fatal/service-runner/unhandled', err);
+                    this.inLogger = false;
+                }
+            };
+
+            this.errorHandler = (err, failedStream) => {
```

Comment:
On the countrary to the previous comment, this is internal dealings of the logger itself, so it's correct that it was left alone here. But why was the logic of adding/removing it was moved up? it should stay here as it was. The logger has a _close method which would be the perfect spot for removing any internal handlers the logger sets up for itself. 

Plus, it's a private matter, let's prefix the field name with `_`

Reply:
Ok

Reply:
This is still not done.

Reply:
Done